snapcraft/_baseplugin.py

```diff
@@ -131,6 +131,10 @@ def clean_build(self):
         """Clean the artifacts that resulted from building this part."""
         pass
 
+    def get_manifest(self):
+        """Return the information to record after the build of this part."""
```

Comment:
Yes, that would be very good.
We haven't finished discussing about docstrings. How do you imagine this? As prose, or just an `rtype` sphinx tag?
I added `:rtype: dict`, let me know what do you think.